### PR TITLE
Create bitwise_addition.java

### PR DESCRIPTION
**Pull request by m-e-r-l-i-n**
Fixes #

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
-  added bitwise addition in java
- 

#### Languages Used:
- java
- 

#### Files Added:
- Bitwise_Addition/bitwise_addition.java
- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by m-e-r-l-i-n** [comment]
I added a file(Bitwise_Addition/bitwise_addition.java). Please check.

File: Bitwise_Addition/bitwise_addition.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    {
+        BufferedReader bu=new BufferedReader(new InputStreamReader(System.in));
+        StringBuilder sb=new StringBuilder();
+        System.out.print("Enter two numbers: ");
+        String s[]=bu.readLine().split(" ");
+        int a=Integer.parseInt(s[0]),b=Integer.parseInt(s[1]);
+        sb.append("\nusing bitwise operations,\n");
+        sb.append("\n"+a+" + "+b+" = "+bitwise(a,b));
+        System.out.print(sb);
+    }
```
Comment (atarax665) [inline Bitwise_Addition/bitwise_addition.java:14]: Leave a line after end of every function

File: Bitwise_Addition/bitwise_addition.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    static int bitwise(int a,int b)
+    {
+        while(b!=0)
```
Comment (atarax665) [inline Bitwise_Addition/bitwise_addition.java:17]: Give space before and after the operators.

File: Bitwise_Addition/bitwise_addition.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        {
+            int carry=a&b;  //common set bits of a and b
+            a=a^b;  //sum of bits of a and b where at least one bit is unset
+            b=carry<<1; //left shifting carry by 1 so that a can give the sum
+        }
+        return a;
+    }
+}
+/*
```
Comment (atarax665) [inline Bitwise_Addition/bitwise_addition.java:26]: Leave a line here.

**Comment by m-e-r-l-i-n** [comment]
updated...please check

File: Bitwise_Addition/bitwise_addition.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
+import java.io.*;
```
Comment (atarax665) [inline Bitwise_Addition/bitwise_addition.java:1]: Add a heading with file name

Comment (atarax665) [reply Bitwise_Addition/bitwise_addition.java:1]: And squash your commits when done.

**Comment by m-e-r-l-i-n** [comment]
please check updates...

**Comment by asha15** [comment]
@m-e-r-l-i-n  can i know the issue number?

**Comment by m-e-r-l-i-n** [comment]
issue number #2965

**Comment by asha15** [comment]
Due to not having an assigned issue , I'm closing this PR

**Pull request closed without merging**